rpc/src/client/channel.rs

```diff
@@ -155,19 +139,17 @@ impl<Req, Resp> Channel<Req, Resp> {
 
 /// A server response that is completed by request dispatch when the corresponding response
 /// arrives off the wire.
+#[pin_project(PinnedDrop)]
 #[derive(Debug)]
 struct DispatchResponse<Resp> {
     response: Timeout<oneshot::Receiver<Response<Resp>>>,
+    #[pin]
     ctx: context::Context,
```

Comment:
I suspect I did this wrong originally. `context::Context` impls `Unpin` right? We may be able to leave this out.